components/homme/src/share/compose/compose_cedr_sl_run_global.cpp

```diff
@@ -5,6 +5,76 @@
 namespace homme {
 namespace sl {
 
+template <int np_, typename MT>
+void run_relaxed_local (CDR<MT>& cdr, const Data& d, Real* q_min_r,
```

Comment:
Can you add some comment (inline or at the top of the fcn) regarding what's happening in here?

Reply:
Perhaps also for the run_global fcn. I am not familiar with the algo, which doesn't help, but the code is a bit dense, which makes it hard to read anyways. Some comments (even just a big picture, not necessarily fine-grained) may help those who one day may have to maintain this.

Reply:
I'll remove these CEDR additions. They correspond to Sect. 2.2.4 of https://gmd.copernicus.org/articles/15/6285/2022/gmd-15-6285-2022.html. I had these lying around and unwisely added them to this already lengthy PR.

Reply:
Mods pushed.